sale_order_type/models/sale.py

```diff
@@ -151,3 +188,18 @@ def _prepare_invoice(self):
         if self.type_id:
             res["sale_type_id"] = self.type_id.id
         return res
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    route_id = fields.Many2one(compute="_compute_route_id", store=True, readonly=False)
+
+    @api.depends("order_id.type_id")
```

Comment:
I think this field is not correctly resolved in the one2many. Maybe you need an extra `order_type_id` helper computed field.